forge-game/src/main/java/forge/game/card/Card.java

```diff
@@ -5987,9 +6000,21 @@ public final boolean sharesNameWith(final String name) {
         boolean shares = getName(true).equals(name);
 
         // Split cards has extra logic to check if it does share a name with
-        if (isSplitCard()) {
-            shares |= name.equals(getState(CardStateName.LeftSplit).getName());
-            shares |= name.equals(getState(CardStateName.RightSplit).getName());
+        if (!shares) {
+            if (isInPlay()) {
```

Comment:
ugh, this two names stuff is really confusing me...
How do they get overwritten?
Here's another case:
![image](https://github.com/user-attachments/assets/2bf08801-8122-406f-a884-1986d27c4c60)

- both Rooms must be unlocked to only the same side
- then animate one to enchant it
- if you target the creature with the Boardwipe the room gets destroyed too
- but if you target the room the creature lives

something can't be right there 🤔

Reply:
Ugh, I probably need to debug that more

Reply:
ugh, can you help me to debug this?

Reply:
if the Room is enchanted with _Witness Protection_ it will only have one name instead of two?

Reply:
should be ok for now I think